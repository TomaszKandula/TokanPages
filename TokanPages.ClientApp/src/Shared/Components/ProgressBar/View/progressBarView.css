.progress-bar-box {
    display: flex;
    align-items: center;
    justify-content: center;
}

.progress-bar-loader {
    border-bottom-color: transparent !important;
    border-radius: 50%;
    display: inline-block;
    box-sizing: border-box;
    animation: rotation 1s linear infinite;
}

@keyframes rotation {
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
} 