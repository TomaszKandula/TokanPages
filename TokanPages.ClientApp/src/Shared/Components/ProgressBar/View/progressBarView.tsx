import * as React from "react";
import { Box, CircularProgress } from "@material-ui/core";

interface ProgressBarViewProps {
    styleObject?: object;
    size?: number;
}

export const ProgressBarView = (props: ProgressBarViewProps): React.ReactElement => {
    const externalStyle = props.styleObject !== undefined ? props.styleObject : {};
    return (
        <Box className="progress-bar-box" style={externalStyle}>
            <Box m="auto">
                <CircularProgress className="progress-bar-progress" size={props.size} />
            </Box>
        </Box>
    );
};
