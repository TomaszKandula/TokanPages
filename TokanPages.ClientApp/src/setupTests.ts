// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
