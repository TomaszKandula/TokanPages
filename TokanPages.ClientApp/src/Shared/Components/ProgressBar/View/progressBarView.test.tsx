import "../../../../setupTests";
import React from "react";
import { render } from "@testing-library/react";
import { ProgressBarView } from "./progressBarView";

describe("test progress bar component", () => {
    it("should correctly render 'CenteredCircularLoader' component.", () => {
        const html = render(<ProgressBarView />);
        expect(html).toMatchSnapshot();
    });
});
