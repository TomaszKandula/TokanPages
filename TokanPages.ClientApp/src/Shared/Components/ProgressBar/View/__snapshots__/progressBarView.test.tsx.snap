// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test progress bar component should correctly render 'CenteredCircularLoader' component. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-1 progress-bar-box"
      >
        <div
          class="MuiBox-root MuiBox-root-2"
        >
          <div
            class="MuiCircularProgress-root progress-bar-progress MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-1 progress-bar-box"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <div
          class="MuiCircularProgress-root progress-bar-progress MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
